std/ascii.d

```diff
@@ -491,6 +491,39 @@ unittest
     assert(!isASCII(128));
 }
 
+/**
+* Params:
+*     s = The string to test
+* Returns:
+*     Whether or not all of `s` is in the ASCII character set - i.e. in the
+*     range `0 .. 0x7F`.
+*/
+bool isASCII(S)(S s) if (isSomeString!S)
+{
+    import std.string : representation;
+
+    // avoid auto-decoding overhead
+    foreach (item; s.representation)
+        if (!isASCII(item))
```

Comment:
Yes, @JackStouffer move to `return s.representation.all!isASCII();`. This will create less template bloat